docs/attributes-registry/db.md

```diff
@@ -26,13 +26,14 @@
 | `db.instance.id` | string | An identifier (address, unique name, or any other identifier) of the database instance that is executing queries or mutations on the current connection. This is useful in cases where the database is running in a clustered environment and the instrumentation is able to record the node executing the query. The client may obtain this value in databases like MySQL using queries like `select @@hostname`. | `mysql-e26b99z.example.com` |
 | `db.name` | string | This attribute is used to report the name of the database being accessed. For commands that switch the database, this should be set to the target database (even if the command fails). [1] | `customers`; `main` |
 | `db.operation` | string | The name of the operation being executed, e.g. the [MongoDB command name](https://docs.mongodb.com/manual/reference/command/#database-operations) such as `findAndModify`, or the SQL keyword. [2] | `findAndModify`; `HMSET`; `SELECT` |
-| `db.statement` | string | The database statement being executed. | `SELECT * FROM wuser_table`; `SET mykey "WuValue"` |
+| `db.query.parameter.<key>` | string | The query parameters, `<key>` being the parameter name (or index number), the value being the parameter value. | `SELECT * FROM wuser_table`; `SET mykey "WuValue"` |
```

Comment:
Overall I believe it's very useful to have this 👍 

One question I'd raise is this: Is this field relational DB specific or do we plan to have this in general for all database systems?

https://github.com/open-telemetry/semantic-conventions/issues/716 focuses on SQL and for that, this covers what we need, but what about non-relational DBs? In theory e.g. for Elasticsearch or MongoDB the whole json could be parsed into these fields - which I don't think we want to do here.

So I'd suggest:
- If this only applies to SQL then shouldn't this be in some `sql` namespace? There is already [`sql.table`](https://github.com/open-telemetry/semantic-conventions/blob/main/model/registry/db.yaml#L271) - looking at https://github.com/open-telemetry/semantic-conventions/issues/716 what we try to solve here seems to be more closer to a SQL specific thing. Also, in general I think the difference between SQL and non-SQL is big enough to move things that are really relational DB specific into a `sql` namespace to not confuse implementors of non-SQL instrumentations. 
- If this also applies to non-SQL, I think we should have examples for non-SQL here.

Reply:
Following up from today's meeting.

Here is an example for Elasticsearch:

```
curl -X POST "localhost:9200/my-index-000001/_doc/?pretty" -H 'Content-Type: application/json' -d'
{
  "@timestamp": "2099-11-15T13:12:00",
  "message": "GET /search HTTP/1.1 200 1070000",
  "user": {
    "id": "kimchy"
  }
}
```

So my point is: what is the interpretation of `db.query.parameter.<key>` for something like this?

Would it be `db.query.parameter.@timestamp:"2099-11-15T13:12:00"`, `db.query.parameter.message:"GET /search HTTP/1.1 200 1070000"`, `db.query.parameter.user.id:"kimchy" `? I think that'd make sense. But then this could explode very quickly and essentially we just re-create the query - the json here can be very big. To me this seems to be different compared to bind values in SQL. Other interpretation can be that those are not parameters, because it's the actual data that it gets inserted. But then where is the line between parameters and not-parameters in NoSQL?

Sorry if this goes into too much - happy to stop here and I agree for most cases the PR is totally ok.

Reply:
You're referencing the elasticsearch API to [insert data into elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html#create-document-ids-automatically). The question is whether or not this API and its parameters qualify for `db.query.parameter.<key>=value`.

If the answer is yes, then `db.query.parameter.<key>=value` is essentially going to capture every record inserted into a database. One reason why this is problematic is that many databases, including elasticsearch, support complex types. OpenTelemetry attributes do not, so these would need to be stringified before capture.

If no, then why? SQL insert / update statements will be equally noisy, and similarly can include complex types through things like [postgres jsonb](https://www.postgresql.org/docs/current/datatype-json.html).

Maybe we say something like:
- `db.query.parmaeter.<key>` is captured when `db.operation.name = SELECT` or similar operation for reading data from database.
- AND the query is executed in a manner similar to SQL-prepared statement. We want to avoid encoding entire query DSLs (i.e. see elasticsearch [query DSL](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-filter-context.html)) into `db.query.parameter.<key>`. Limit usage to cases where a query is submitted and there are template vars / parameters which are replaced.

Reply:
> > If the answer is yes, then db.query.parameter.=value is essentially going to capture every record inserted into a database.
> 
> Maybe a stupid question as I wasn't in the meetings, but isn't the `query` namespace already making it explicit this is for "select" and not insert operations?

That's a good point, and I think so far we haven't considered this. If we say this is only for `select` (and as @jack-berg suggests, we also say we don't try to parse query DSLs - see Elasticsearch link) then this is solved. However, by looking at https://github.com/open-telemetry/semantic-conventions/issues/716 I'm not sure users would be satisfied with that. My understanding is that these fields originate from the need to capture SQL bind values - those can still be present in `update` or `insert` statements and I don't think users would be ok to only capture those for `select` statements. In the product I work on, users also asked for capturing those values and those discussions were never limited to `select` statements - it was always about bind values in general for all sql statements.

But good point on the `query` namespace. What do others think? Is it really ok to limit this to `select`?

Reply:
Quite a bit is lost if only `select` is applicable for `db.query.*` and `db.statement` is removed.

I think capturing whole statement and/or parameters is nothing one will do on default. On default I would not capture parameters and statements only masked with a reasonable size limit.

Reply:
> Sorry if this goes into too much - happy to stop here and I agree for most cases the PR is totally ok.

no worries, clearly this is something worth clarifying

> `db.query.parameter.<key>` is captured when `db.operation.name` = SELECT or similar operation for reading data from database.

I think `db.query.parameter.<key>` applies to bind parameters for all prepared statement, not only SELECT statements.

> Limit usage to cases where a query is submitted and there are template vars / parameters which are replaced.

maybe this is how we could clarify the situation, and say that `db.query.parameter.<key>` only apply to templatized queries

Reply:
I updated the PR, let me know if this helps to clarify:

> Query parameters should only be captured when `query.text` is parameterized with placeholders.

Reply:
I think this clarifies it 👍